src/sentry/search/events/builder.py

```diff
@@ -386,11 +392,7 @@ def resolve_select(
 
         # Add threshold config alias if there's a function that depends on it
         # TODO: this should be replaced with an explicit request for the project_threshold_config as a column
-        for column in {
-            "apdex()",
-            "count_miserable(user)",
-            "user_misery()",
-        }:
+        for column in self.config.custom_threshold_columns:
```

Comment:
Nit: it might be nice if this can be refactored and moved out of the base query builder entirely and into a dataset-specific one

Reply:
I think the TODO above this line is what will block that, but totally agree :+1: